File: Java/LeetCode/Remove All Occurrences of a Substring.java
// https://leetcode.com/problems/remove-all-occurrences-of-a-substring

class Solution {
  
    public String removeOccurrences(String s, String part) {
        while (s.contains(part)) {
            s = s.replaceFirst(part, "");
        }

        return s;
    }
  
}

// Recursive solution
/*
class Solution {
    public String removeOccurrences(String s, String part) {
        if (!s.contains(part)) {
            return s;
        }
        return removeOccurrences(s.replaceFirst(part, ""), part);
    }
}
*/

File: Kotlin/LeetCode/Remove All Occurrences of a Substring.kt
// https://leetcode.com/problems/remove-all-occurrences-of-a-substring

class Solution {
    fun removeOccurrences(s: String, part: String): String {
        var str = s
        
        while (str.contains(part)) {
            str = str.replaceFirst(part, "")
        }

        return str
    }
}

// Recursive solution
/*
class Solution {
    fun removeOccurrences(s: String, part: String): String {
        if (!s.contains(part)) {
            return s
        } 
        return removeOccurrences(s.replaceFirst(part, ""), part);
    }
}
*/
